src/init-nmos.js

```diff
@@ -70,25 +84,32 @@ function getPrioritised (representations, priority, version, protocol) {
   }
 }
 
-export default (start) => {
-  axios
-    .get(`${queryProtocol}://${parsedUrl.hostname}:${queryPort}/x-nmos/node/${queryVersion}/self/`)
+export function getServiceUrl (serviceType, apiVersion, priority) {
+  return axios
+    .get(`${queryProtocol}://${parsedUrl.hostname}:${queryPort}/x-nmos/node/${apiVersion}/self/`)
     .then(result => {
       let service = result.data.services.filter(service => {
         return service.type.includes('mdnsbridge')
       })[0]
-      let url = service.href + 'nmos-query/'
+      let url = service.href + serviceType + '/'
       return axios.get(url)
     })
     .then(result => {
       let representations = result.data.representation
       if (representations.length === 0) {
-        throw new Error('No Query APIs identified from mDNS Bridge')
+        throw new Error('No ' + serviceType + ' APIs identified from mDNS Bridge')
       }
-      let url = getPrioritised(representations, queryPriority, queryVersion, queryProtocol)
+      let url = getPrioritised(representations, priority, apiVersion, queryProtocol)
       if (url === '') {
-        throw new Error('No URL identified for Query API')
+        throw new Error('No URL identified for ' + serviceType + ' API')
       }
+      return url
+    })
+}
+
+export default (start) => {
+  getServiceUrl('nmos-query', queryVersion, queryPriority)
```

Comment:
I'm not sure the queryVersion here will always do exactly what it suggests as it's used for both the Node API and Query API version within the function. Might just need a rename to be more generic, or the GET to the Node API may need a separate version parameter?